torch/_inductor/scheduler.py

```diff
@@ -2030,6 +2030,25 @@ def can_fuse_vertical(self, node1, node2):
             if node1_names & self.name_to_fused_node[name].ancestors:
                 why("intermediate nodes between node1 & node2")
                 return False
+
+        # similar to can_inplace, if we are going to fuse a write subsequent to a read
+        # require that the indexing and size is the same
+        for write in node2.read_writes.writes:
+            for read in node1.read_writes.reads:
+                if write.name != self.mutation_renames.get(read.name, read.name):
+                    continue
+
```

Comment:
@mlazos that was checking producer reads with consumer writes where as this is checking produce writes with consumer reads but I can refactor the check and reuse it